src/components/flat-table/sort/sort.component.tsx

```diff
@@ -11,10 +13,20 @@ export interface SortProps {
   onClick?: () => void;
   /** Sets the content of `FlatTableSortHeader` */
   children?: React.ReactNode;
+  /**
+   * Creates an accessible name for `FlatTableSortHeader`, it is recommended that the
+   * text content of the `FlatTableSortHeader` as well as the current `sortType` are included
+   * in this accessible name
+   */
+  accessibleName?: string;
 }
 
-const Sort = ({ children, onClick, sortType }: SortProps) => {
+const Sort = ({ children, onClick, sortType, accessibleName }: SortProps) => {
   const id = useRef(guid());
+  const locale = useLocale();
+
+  const verifiedChild = typeof children === "string" ? children : undefined;
```

Comment:
Comment: I wonder if we should just bite the bullet here and make a breaking change so only string values can be passed to `children`. It might be worth doing a bit of an investigation and seeing if anyone has passed a node here instead, I can't see why or how it would have worked if they did but it would at least allow us to get a measure of the impact if we do change it etc